src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -86,10 +86,16 @@ export default function getAdaptationSwitchStrategy(
     return { type: "continue", value: undefined };
   }
   const bufferedRanges = convertToRanges(buffered);
-  const start = period.start;
-  const end = period.end == null ? Infinity :
-                                   period.end;
-  const intersection = keepRangeIntersection(bufferedRanges, [{ start, end }]);
+  const contentStart = period.getContentStart();
+  const contentEnd = period.getContentEnd();
+  if (contentStart === null ||
+      contentEnd === null) {
+    return { type: "continue", value: undefined };
+  }
+  const start = contentStart ?? period.start;
+  const end = contentEnd ?? period.end;
+  const intersection = keepRangeIntersection(bufferedRanges,
+                                             [{ start, end: (end ?? Infinity) }]);
```

Comment:
I'm not sure. What if the real segment's end if a little over the contentEnd yet still below the period's end?

I think limiting by the Period's end makes more sense here.